lib/dependenciesInstall.js

```diff
@@ -1,40 +1,71 @@
+var fs = require('fs');
+var path = require('path');
+
+// This function gets called after every plugin addition, as specified in plugin.xml
 module.exports = function(context) {
+
+    var options = context.opts;
     var Q = context.requireCordovaModule('q');
-    var npm = context.requireCordovaModule('npm');
-    var fs = require('fs');
-    var path = require('path');
 
     // package.json file is a sister folder to 'lib' folder, in which we currently are executing this script from
-    var packageJSONDir = path.join(__dirname, '../'); 
+    var packageJSONDir = path.join(__dirname, '../');
     var packageJSONLocation = path.join(packageJSONDir, 'package.json');
-    var dependencies = JSON.parse(fs.readFileSync(packageJSONLocation, 'utf-8')).dependencies;
-    
+    var packageJSON = JSON.parse(fs.readFileSync(packageJSONLocation, 'utf-8'));
+
+    var pluginName = packageJSON.name;
+
+
+    // This function checks if the pluginSpec provided refers to 'cordova-plugin-liverelad'
+    var isLiveReloadPlugin = function(pluginSpec) {
+
+        // A plugin can be installed via an npm package, a path or a git url. 
+        //     e.g: https://github.com/omefire/cordova-plugin-livereload.git
+        //          https://github.com/schacon/simplegit 
+        //          cordova-plugin-livereload
+        //          ../cordova-plugin-livereload/
+        //          ../plugin/
+        //          C:\Plugins\cordova-plugin-livereload\
+        //          C:\Plugins\plugin\
+        // Instead of testing for all those possibilities,
+        // ... let's test instead for whether the livereload name is contained in the npm package name, the local folder path or the git url
+        return pluginSpec.indexOf(pluginName) != -1;
+    };
+
+    // Only install livereload dependencies if 'cordova-plugin-livereload' is among the plugins we've just installed
+    // ... Otherwise, livereload dependencies get installed after every plugin install
+    if (options.plugins.filter(isLiveReloadPlugin).length === 0) {
+        return Q();
+    }
+
+    var dependencies = packageJSON.dependencies;
+
     var depNames = [];
-    for(var dep in dependencies) {
-    	if (dependencies.hasOwnProperty(dep)) {
+    for (var dep in dependencies) {
```

Comment:
Not related to this change - why do we parse the package.json file and put the dependencies here? Instead why don't we just do a full `npm install` of the package.json itself?

npm install does more than just call npm install on dependencies - there is bundledepdencies & peer dependencies - blah blah - we should really not try to do much here. Of course, as we look to use shrinkwrapping this might change as well.